forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -7950,26 +7950,14 @@ public void addChosenModes(SpellAbility ability, String mode, boolean yourCombat
                 }
             }
         } else {
-            List<String> result = chosenModesTurn.get(original);
-            if (result == null) {
-                result = Lists.newArrayList();
-                chosenModesTurn.put(original, result);
-            }
+            List<String> result = chosenModesTurn.computeIfAbsent(original, k -> Lists.newArrayList());
```

Comment:
k -> Lists.newArrayList() can't be replaced with `Lists::newArrayList` or something, right?

Reply:
```
java: incompatible types: bad return type in lambda expression
    inference variable E has incompatible bounds
      equality constraints: java.lang.String
      lower bounds: forge.game.spellability.SpellAbility
```
Apparently not. It's trying to plug the key into the parameters of newArrayList, but that would create an `ArrayList<SpellAbility>` when the map wants an `ArrayList<String>`.

Reply:
ah yeah because it is a Function and not a Supplier